sql/core/src/main/scala/org/apache/spark/sql/execution/python/EvaluatePython.scala

```diff
@@ -198,7 +198,7 @@ object EvaluatePython {
 
     case udt: UserDefinedType[_] => makeFromJava(udt.sqlType)
 
-    case other => (obj: Any) => nullSafeConvert(other)(PartialFunction.empty)
+    case other => (obj: Any) => nullSafeConvert(obj)(PartialFunction.empty)
```

Comment:
Hi, @sarutak . Can we have a test case for this?

Reply:
+1

Reply:
btw, for any input, this converter in this pattern returns null? I mean we cannot write it here like this? 
```
case other => (_: Any) => null
```

Reply:
> btw, for any input, this converter in this pattern returns null?
Yes so it's difficult to test based on the return value. 
Or, how about testing that the converter still returns null, meaning there is no regression?

Reply:
either way seems fine to me